packages/chart/src/ChartUtils.ts

```diff
@@ -552,6 +565,65 @@ class ChartUtils {
     return title;
   }
 
+  static getTimeZoneDiff(
+    calendarTimeZone: DhType.i18n.TimeZone,
+    formatterTimeZone?: DhType.i18n.TimeZone
+  ): number {
+    return formatterTimeZone
+      ? (calendarTimeZone.standardOffset - formatterTimeZone.standardOffset) /
+          60
+      : 0;
+  }
+
+  /**
+   * Creates closed periods for a partial holiday.
+   *
+   * @param holidayPeriods the business periods for the holiday
+   * @param calendarPeriods the business periods for the calendar
+   * @returns an array of closed periods for the partial holiday. Should be the periods during the regular business hours that are _not_ specified by the holiday periods.
+   */
+  static createClosedPeriodsForPartialHoliday(
```

Comment:
This takes periods and returns decimal ranges, maybe rename to `createClosedRangesFor...`?
```suggestion
   * @returns an array of closed ranges for the partial holiday. Should be the ranges during the regular business hours that are _not_ specified by the holiday periods.
   */
  static createClosedRangesForPartialHoliday(
```